Frends.Odoo.Request/Frends.Odoo.Request.Tests/UnitTests.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using dotenv.net;
+
+namespace Frends.Odoo.Request.Tests;
+
+using Frends.Odoo.Request.Definitions;
+using NUnit.Framework;
+
+[TestFixture]
+internal class UnitTests
+{
+    private readonly string _odooUrl = Environment.GetEnvironmentVariable("ODOO_URL");
+    private readonly string _username = Environment.GetEnvironmentVariable("ODOO_USERNAME");
+    private readonly string _password = Environment.GetEnvironmentVariable("ODOO_PASSWORD");
+    private readonly string _database = Environment.GetEnvironmentVariable("ODOO_DATABASE");
+
+    [Test]
+    public async Task Request_InvalidCredentials_Error()
+    {
+        // Arrange
+        var input = GetDefaultInput();
+        var options = GetDefaultOptions();
+        options.Password = "invalid";
+
+        // Act
+        var result = await Odoo.Request(input, options, default);
+
+        // Assert
+        Assert.IsFalse(result.Success);
+        Assert.IsTrue(result.Error.StartsWith("Authentication failed. Odoo's response: Access Denied"), result.Error);
```

Comment:
what would be the difference if we introduce a const string though? you would navigate to failed test and then also have to navigate to the const instead of just having the value directly in the test itself.

Reply:
Imo the difference is, that in this approach if we want to change string with info we need to do it in two places (or maybe more if we create more tests in the future or any other place we will want to get this string), when string will be in const it will change always only in one place